tests/unit/test_control.py

```diff
@@ -99,12 +105,17 @@ def start_injecting(device: str, preset: str):
             start_history.append((device, preset))
             daemon.injectors[device] = Injector()
 
-        daemon.start_injecting = start_injecting
+        patch.object(daemon, "start_injecting", start_injecting).start()
 
         global_config.set_autoload_preset(groups_[0].key, presets[0])
         global_config.set_autoload_preset(groups_[1].key, presets[1])
 
-        communicate(options("autoload", None, None, None, False, False, False), daemon)
+        self.input_remapper_control.communicate(
```

Comment:
todo: ditch the communicate method, call desired method directly (like .autoload, .start, etc.)